docarray/array/storage/memory/seqlike.py

```diff
@@ -57,4 +59,6 @@ def __add__(self, other: Union['Document', Sequence['Document']]):
 
     def extend(self, values: Iterable['Document']) -> None:
         self._data.extend(values)
-        self._rebuild_id2offset()
+        # extend _id_to_index, no rebuilt needed
+        last_idx = len(self._id_to_index)
+        self._id_to_index.update({d.id: i + last_idx for i, d in enumerate(values)})
```

Comment:
Let's also handle the case where the `_id_to_index` is not rebuilt.
I guess this should take care of it:
```suggestion
        # extend _id_to_index, no rebuilt needed
        last_idx = len(self._id2offset)
        self._id_to_index.update({d.id: i + last_idx for i, d in enumerate(values)})
```

Reply:
completely agree